plugins/item_previews/web_client/views/ItemPreviewWidget.js

```diff
@@ -1,103 +1,175 @@
+        }
+    },
+
+    isNotFull: function () {
+        return this.$('.g-widget-item-prevews-wrapper').height() <= this.$('.g-widget-item-previews-container').height();
+    },
+
+    allItemAdded: function () {
+        return this.renderedIndex === this.supportedItems.length;
+    },
+
+    addMoreItem: function () {
+        this.addingMoreItems = true;
+        var items = this.supportedItems.slice(this.renderedIndex, this.renderedIndex + this._LOAD_BATCH_SIZE);
+        this.renderedIndex = Math.min(this.renderedIndex + this._LOAD_BATCH_SIZE, this.supportedItems.length);
+
+        if (!items.length) {
+            return;
+        }
 
-            // Ajax request the JSON files to display them.
-            restRequest({
-                path: 'item/' + id + '/download',
-                type: 'GET',
-                error: null // don't do default error behavior (validation may fail)
-            }).done(function (resp) {
-                view.$('.json[data-id="' + id + '"]').text(JSON.stringify(resp, null, '\t'));
-            }).fail(function (err) {
-                console.error('Could not preview item', err);
+        var $container = this.$('.g-widget-item-prevews-wrapper');
+        _whenAll(items.map((item) => {
+            return restRequest({
+                path: `item/${item._id}/files`
+            }).then((files) => {
+                return _whenAll(files.map((file) => {
+                    return this.tryGetFileContent(file, file.mimeType, 'file');
+                }));
+            });
+        })).then((results) => {
+            items.forEach((item, i) => {
+                var contents = results[i].filter((result) => result);
+                if (contents.length) {
+                    $container.append(ItemPreviewItemTemplate({
+                        item: item,
+                        contents: contents
+                    }));
+                }
             });
+            this.addingMoreItems = false;
+
+            // In case the container is not fully filled add more items
+            if (!this.allItemAdded() && this.isNotFull()) {
+                this.addMoreItem();
+            }
+            return undefined;
         });
+    },
+
+    isNearBottom: function () {
+        var $container = this.$('.g-widget-item-previews-container');
+        return $container.scrollTop() + $container.innerHeight() >= $container[0].scrollHeight - 200;
+    },
+
+    onScroll: function (e) {
+        if (this.isNearBottom()) {
+            if (!this.addingMoreItems && !this.allItemAdded()) {
+                this.addMoreItem();
+            }
+        }
+    },
 
-        this.$('.g-widget-item-previews-wrap').tooltip({
-            container: this.$el,
-            placement: 'left',
-            animation: false,
-            delay: {show: 100}
+    tryGetFileContent: function (file, contentType, type) {
+        if (this._isJSONItem(file.name) && contentType === 'application/octet-stream') {
```

Comment:
It depends on how you upload.  If you upload through the browser, you should get `application/json`.  In any case, the correct mime type is `application/json`.

Since you are using the file extension to detect the type for both images and json, this extra condition is unnecessary.  I would just remove it.